tests/test_hooks/test_prepare_tta_hook.py

```diff
@@ -0,0 +1,130 @@
+        self.linear = torch.nn.Linear(1, 1)
+
+    def forward(self, inputs, data_samples, mode='tensor'):
+        return data_samples
+
+
+class ToyTestTimeAugModel(BaseTTAModel):
+
+    def merge_preds(self, data_samples_list):
+        result = [sum(x) for x in data_samples_list]
+        return result
+
+
+class ToyTTAPipeline:
+
+    def __call__(self, result):
+        return {key: [value] for key, value in result.items()}
+
+
+class TestPrepareTTAHook(RunnerTestCase):
+
+    def setUp(self) -> None:
+        super().setUp()
+        TRANSFORMS.register_module(module=ToyTTAPipeline, force=True)
+        MODELS.register_module(module=ToyModel, force=True)
+        MODELS.register_module(module=ToyTestTimeAugModel, force=True)
+        DATASETS.register_module(module=ToyDatasetTTA, force=True)
+
+    def tearDown(self):
+        super().tearDown()
+        TRANSFORMS.module_dict.pop('ToyTTAPipeline', None)
+        MODELS.module_dict.pop('ToyModel', None)
+        MODELS.module_dict.pop('ToyTestTimeAugModel', None)
+        DATASETS.module_dict.pop('ToyDatasetTTA', None)
+
+    def test_init(self):
+        tta_cfg = dict(type='ToyTTAModel')
+        prepare_tta_hook = PrepareTTAHook(tta_cfg)
+        self.assertIsInstance(prepare_tta_hook, Hook)
+        self.assertIs(tta_cfg, prepare_tta_hook.tta_cfg)
+
+    def test_before_test(self):
+        # Test with epoch based runner.
+        cfg = copy.deepcopy(self.epoch_based_cfg)
+        cfg.custom_hooks.append(
+            dict(
+                type='PrepareTTAHook',
+                tta_cfg=dict(type='ToyTestTimeAugModel')))
+        cfg.model = dict(type='ToyModel')
+        cfg.test_dataloader.dataset = dict(
+            type='ToyDatasetTTA', pipeline=dict(type='ToyTTAPipeline'))
+        runner = self.build_runner(cfg)
+        self.assertNotIsInstance(runner.model, BaseTTAModel)
+        runner.test()
+        self.assertIsInstance(runner.model, BaseTTAModel)
+
+        # Test with iteration based runner
+        cfg = copy.deepcopy(self.iter_based_cfg)
+        cfg.custom_hooks.append(
+            dict(
+                type='PrepareTTAHook',
+                tta_cfg=dict(type='ToyTestTimeAugModel')))
+        cfg.model = dict(type='ToyModel')
+        cfg.test_dataloader.dataset = dict(
+            type='ToyDatasetTTA', pipeline=dict(type='ToyTTAPipeline'))
+        runner = self.build_runner(cfg)
+        self.assertNotIsInstance(runner.model, BaseTTAModel)
+        runner.test()
+        self.assertIsInstance(runner.model, BaseTTAModel)
+
+        # Test with ddp
+        if torch.cuda.is_available() and torch.distributed.is_nccl_available():
+            self.setup_dist_env()
+            cfg.launcher = 'pytorch'
+            runner = self.build_runner(cfg)
+            self.assertNotIsInstance(runner.model, BaseTTAModel)
+            runner.test()
+            self.assertIsInstance(runner.model, BaseTTAModel)
+
+
+class TestBuildRunenrWithTTA(TestPrepareTTAHook):
```

Comment:
Is it necessary to inherit from `TestPrepareTTAHook`?

Reply:
It is necessary, otherwise, it needs to register tta related module itself.

Reply:
But many tests are not necessary for `TestBuildRunenrWithTTA`.